connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -143,6 +162,70 @@ public ConnectorStateInfo connectorStatus(String connName) {
                 status.workerId(), status.trace());
     }
 
+
+    @Override
+    public ConfigInfos validateConfigs(String connType, Map<String, String> connectorConfig) {
+        ConfigDef connectorConfigDef = ConnectorConfig.configDef();
+        List<ConfigValue> connectorConfigValues = connectorConfigDef.validate(connectorConfig);
+        ConfigInfos result = generateResult(connType, connectorConfigDef.configKeys(), connectorConfigValues);
+
+        if (result.errorCount() != 0) {
+            return result;
+        }
+
+        ConnectorConfig connConfig = new ConnectorConfig(connectorConfig);
+        String connName = connConfig.getString(ConnectorConfig.NAME_CONFIG);
+        Connector connector = getConnector(connType, connName);
+
+        Map<String, ConfigKey> configKeys = connector.configDef().configKeys();
+        List<ConfigValue> configValues = connector.validate(connectorConfig);
+        configKeys.putAll(connectorConfigDef.configKeys());
```

Comment:
The validation takes the configKeys originally defined in the connector ConfigDef. We only add the common configurations for all connectors when returning the result. May be we need a copy of configKeys here?